helm/druid/templates/broker/role.yaml

```diff
@@ -0,0 +1,21 @@
+{{- if .Values.rbac.create }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ template "druid.broker.fullname" . }}
+  labels:
+    app: {{ include "druid.name" . }}
+    chart: {{ include "druid.chart" . }}
+    component: {{ .Values.broker.name }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+      - configmaps
+    verbs:
+      - '*'
```

Comment:
Unclear on why the broker needs access to delete Pods, or really any of these. Are these permissions only needed by the Overlord?

Reply:
I found that when this role did not exist none of the pods could start.  They seem to all manage labels when using the kubernetes extension.  Unclear if they will remove when not needed the same as they add when needed.  I can test that and narrow down permission sets for each service.

Reply:
I pushed up the changes requested except for the RBAC changes since I haven't actually been able to say I've had a successful deployment with these chart changes yet.

Reply:
Turns out this is a bug with the kubernetes overlord extension and kubernetes 1.25+

Reply:
Were you able to file an issue for it?

Reply:
Discovered this with the author of the kubernetes overlord extension @churromorales .  I believe https://github.com/apache/druid/pull/13759 already fixes the issues and relates to https://github.com/apache/druid/issues/13749 but maybe @churromorales can comment further if another issue is required I will happily make one.

Reply:
@dampcake putting aside the potential for a new issue to be created I would ask that we merge the RBAC in its current state since it matches the official documentation.  Once the bugs are fixed for K8s 1.25 I will test for the minimum viable role permissions and make a new PR to apply those via the helm chart as well as allow for concatenating or overriding the role permissions via the chart `.Values.rbac` map so people won't get blocked on changes required for rbac in the future.

If you're good with this let me know and I'll create the issue and it can be assigned to me.